editor/src/components/canvas/dom-lookup.ts

```diff
@@ -165,32 +186,103 @@ export function findFirstParentWithValidElementPathInner(
         target.parentElement,
         allowDescendantsFromPath,
         parentSceneValidPathsCache,
+        metadata,
+        point,
       )
     }
   }
 }
 
+function extendValidPathsWithNonDomParent(
+  domPath: {
+    static: string
+    staticPath: ElementPath
+    dynamic: ElementPath
+  },
+  nonDomPath: ElementPath,
+): ElementPath | null {
+  // check if there is a dom path which is a descendant of a non-dom path: then the
+  // non-dom path should be a valid target too
+  if (EP.isDescendantOfOrEqualTo(domPath.staticPath, nonDomPath)) {
+    const depthDiff = EP.fullDepth(domPath.staticPath) - EP.fullDepth(nonDomPath)
+    return EP.nthParentPath(domPath.dynamic, depthDiff)
```

Comment:
This is a bit confusing... We've determined that A is a descendant of B, then calculated the difference in the depths of those to be N, and then are moving up the hierarchy N steps from A... So they why is the result not just B? Or is it B?

Reply:
Oh, because it is potentially a dynamic path to an instance of B! Ok can you add a comment about that please